### PR TITLE
refactor(ts): make processImports logic more verbose

### PR DESCRIPTION
**Pull request by bartlomieju**
This is just preparation for rewriting `processImports()` in Rust - I really needed to untangle it a bit and understand the logic.

@kitsonk PTAL

### CODE REVIEW

File: cli/js/compiler.ts [hunk 1]
```diff
@@ -223,10 +223,32 @@ function getExtension(fileName: string, mediaType: MediaType): ts.Extension {
...
+interface SourceFileSpecifierMap {
+  original: string;
+  mapped: string;
+}
+
+/** A global cache of module source files that have been loaded.
+ * This cache will be rewritten to be populated on compiler startup
+ * with files provided from Rust in request message.
+ */
+const SOURCE_FILE_CACHE: Map<string, SourceFile> = new Map();
```
Comment (ry) [inline cli/js/compiler.ts:242]: Maybe can be removed without perf hit

File: cli/js/compiler.ts [hunk 2]
```diff
@@ -223,10 +223,32 @@ function getExtension(fileName: string, mediaType: MediaType): ts.Extension {
...
+/** A map of maps which cache resolved specifier for each import in a file.
+ * This cache is used so `resolveModuleNames` ops is called as few times
+ * as possible.
+ *
+ * First map's key is "referrer" URL ("file://a/b/c/mod.ts")
+ * Second map's key is "raw" import specifier ("./foo.ts")
+ * Second map's value is resolved import URL ("file:///a/b/c/foo.ts")
+ */
+const RESOLVED_SPECIFIER_CACHE: Map<string, Map<string, string>> = new Map();
```
Comment (ry) [inline cli/js/compiler.ts:251]: Maybe can be removed. Op call is very cheap relative to tsc

File: cli/js/compiler.ts (hunk 1 above)
Comment (bartlomieju) [reply cli/js/compiler.ts:242]: Please read the docstring

File: cli/js/compiler.ts (hunk 2 above)
Comment (bartlomieju) [reply cli/js/compiler.ts:251]: It's actually used for handling of type directives

**Pull request merged**